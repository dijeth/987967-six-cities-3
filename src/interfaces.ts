import { id, roomType, coord } from "./types";

export interface OfferMini {
    readonly id: id;
    title: string;
    type: roomType;
    pictures: Array<string>;
    cost: number;
    rating: number;
    isPremium: boolean;
    isFavorite: boolean;
    city: string;
    coord: coord;
};

export interface Offer extends OfferMini {
    bedroomCount: number;
    adultsCount: number;
    insideFeatures: Array<string>;
    userName: string;
    userPicture: string;
    isSuperUser: boolean;
    descriptionTitle: string;
    description: string;
    zoom: number;
}

export interface City {
    name: string;
    zoom: number;
    centerCoord: coord;
}

export interface UserReview {
    readonly id: id;
    userName: string;
    userPicture: string;
    rating: number;
    description: string;
    date: string;
}