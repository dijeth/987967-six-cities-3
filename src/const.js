const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export {OfferType, CITIES};
