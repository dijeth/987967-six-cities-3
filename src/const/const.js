"use strict";
exports.__esModule = true;
exports.OfferType = {
    APARTMENT: "Apartment",
    PRIVATE_ROOM: "Private room"
};
exports.CITIES = [
    "Paris",
    "Cologne",
    "Brussels",
    "Amsterdam",
    "Hamburg",
    "Dusseldorf"
];
exports.InsideFeature = {
    WI_FI: "Wi-Fi",
    WASHING_MACHINE: "Washing machine",
    TOWELS: "Towels",
    HEATING: "Heating",
    COFFEE_MACHINE: "Coffee machine",
    BABY_SEAT: "Baby seat",
    KITCHEN: "Kitchen",
    DISHWASHER: "Dishwasher",
    CABEL_TV: "Cabel TV",
    FRIDGE: "Fridge"
};
exports.ScreenType = {
    MAIN: "MAIN",
    PROPERTY: "PROPERTY"
};
exports.BREAK_STRING = "\n";
exports.MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
exports.MAX_CITY_COUNT = 6;
var SortType;
(function (SortType) {
    SortType["POPULAR"] = "Popular";
    SortType["PRICE_LOW_TO_HIGH"] = "Price: low to high";
    SortType["PRICE_HIGH_TO_LOW"] = "Price: high to low";
    SortType["TOP_RATED_FIRST"] = "Top rated first";
})(SortType = exports.SortType || (exports.SortType = {}));
exports.SORT_LIST = [
    SortType.POPULAR,
    SortType.PRICE_LOW_TO_HIGH,
    SortType.PRICE_HIGH_TO_LOW,
    SortType.TOP_RATED_FIRST
];
exports.AuthorizationStatus = {
    NO_AUTH: "NO_AUTH",
    AUTH: "AUTH"
};
exports.AppRoute = {
    getLogin: function () { return "/login"; },
    getOffer: function (id) { return "/offer/" + id; },
    getFavorites: function () { return "/favorites"; },
    getRoot: function () { return "/"; }
};
exports.ServerRoute = {
    getLogin: function () { return "/login"; },
    getFavorites: function () { return "/favorite"; },
    postFavorites: function (id, status) { return "/favorite/" + id + "/" + status; },
    getNearby: function (id) { return "/hotels/" + id + "/nearby"; },
    getHotels: function () { return "/hotels"; },
    getComments: function (id) { return "/comments/" + id; }
};
exports.MAX_IMAGE_COUNT = 6;
exports.ReviewLength = {
    MIN: 50,
    MAX: 300
};
exports.ServerError = {
    UNAUTHORIZED: 401,
    BAD_REQUEST: 400,
    NOT_FOUND: 404
};
var PlaceCardType;
(function (PlaceCardType) {
    PlaceCardType["DEFAULT"] = "DEFAULT";
    PlaceCardType["FAVORITE"] = "FAVORITE";
    PlaceCardType["NEARBY"] = "NEARBY";
})(PlaceCardType = exports.PlaceCardType || (exports.PlaceCardType = {}));
exports.EMPTY_REVIEW = {
    text: "",
    rating: 0,
    offerID: null
};
exports.BASE_URL = "https://htmlacademy-react-3.appspot.com/six-cities";
exports.TIME_OUT = 5000;
