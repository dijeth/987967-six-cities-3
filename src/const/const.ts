export const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const InsideFeature = {
  WI_FI: `Wi-Fi`,
  WASHING_MACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEE_MACHINE: `Coffee machine`,
  BABY_SEAT: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABEL_TV: `Cabel TV`,
  FRIDGE: `Fridge`
};

export const ScreenType = {
  MAIN: `MAIN`,
  PROPERTY: `PROPERTY`
};

export const BREAK_STRING = `\n`;

export const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const MAX_CITY_COUNT = 6;

export enum SortType {
  POPULAR = `Popular`,
  PRICE_LOW_TO_HIGH = `Price: low to high`,
  PRICE_HIGH_TO_LOW = `Price: high to low`,
  TOP_RATED_FIRST = `Top rated first`,
};

export const SORT_LIST = [
  SortType.POPULAR,
  SortType.PRICE_LOW_TO_HIGH,
  SortType.PRICE_HIGH_TO_LOW,
  SortType.TOP_RATED_FIRST
];

export const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

export const AppRoute = {
  getLogin: () => `/login`,
  getOffer: (id) => `/offer/${id}`,
  getFavorites: () => `/favorites`,
  getRoot: () => `/`,
};

export const ServerRoute = {
  getLogin: () => `/login`,
  getFavorites: () => `/favorite`,
  postFavorites: (id, status) => `/favorite/${id}/${status}`,
  getNearby: (id) => `/hotels/${id}/nearby`,
  getHotels: () => `/hotels`,
  getComments: (id) => `/comments/${id}`,
};

export const MAX_IMAGE_COUNT = 6;

export const ReviewLength = {
  MIN: 50,
  MAX: 300
};

export const ServerError = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404
};

export enum PlaceCardType {
  DEFAULT = `DEFAULT`,
  FAVORITE = `FAVORITE`,
  NEARBY = `NEARBY`,
};

export const EMPTY_REVIEW = {
  text: ``,
  rating: 0,
  offerID: null,
};

export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;
export const TIME_OUT = 5000;
