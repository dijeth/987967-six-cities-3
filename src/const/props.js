import PropTypes from 'prop-types';

export const offerPropType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  pictures: PropTypes.arrayOf(PropTypes.string).isRequired,
  cost: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  isPremium: PropTypes.bool.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  city: PropTypes.string.isRequired,
  coord: PropTypes.arrayOf(PropTypes.number).isRequired
});

export const cityPropType = PropTypes.shape({
  name: PropTypes.string.isRequired,
  zoom: PropTypes.number.isRequired,
  centerCoord: PropTypes.arrayOf(PropTypes.number)
});
