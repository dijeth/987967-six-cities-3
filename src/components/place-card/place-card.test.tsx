import * as React from 'react';
import * as renderer from 'react-test-renderer';
import PlaceCard from './place-card';
import {BrowserRouter} from 'react-router-dom';
import {PlaceCardType} from '../../const/const';
import { OfferMini } from '../../interfaces.js';

const mock: OfferMini = {
  id: `id-1`,
  title: `title-1`,
  type: `apartment`,
  pictures: [`picture-1`],
  cost: 123,
  rating: 4.8,
  isPremium: false,
  isFavorite: true,
  city: `Amsterdam`,
  coord: [123, 456]
};

describe(`<PlaceCard /> should be render correctly`, () => {
  it(`with type === "DEFAULT"`, () => {
    const card = renderer.create(
        <BrowserRouter>
          <PlaceCard
            offer={mock}
            onHover={() => {}}
            type={PlaceCardType.DEFAULT}
            isAuth={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(card).toMatchSnapshot();
  });

  it(`with type === "FAVORITE"`, () => {
    const card = renderer.create(
        <BrowserRouter>
          <PlaceCard
            offer={mock}
            onHover={() => {}}
            type={PlaceCardType.FAVORITE}
            isAuth={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(card).toMatchSnapshot();
  });

  it(`with type === "NEARBY"`, () => {
    const card = renderer.create(
        <BrowserRouter>
          <PlaceCard
            offer={mock}
            onHover={() => {}}
            type={PlaceCardType.NEARBY}
            isAuth={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(card).toMatchSnapshot();
  });

  it(`with isPremium === true`, () => {
    mock.isPremium = true;
    const card = renderer.create(
        <BrowserRouter>
          <PlaceCard
            offer={mock}
            onHover={() => {}}
            type={PlaceCardType.DEFAULT}
            isAuth={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(card).toMatchSnapshot();
  });

  it(`with isFavorite === false`, () => {
    mock.isFavorite = false;
    const card = renderer.create(
        <BrowserRouter>
          <PlaceCard
            offer={mock}
            onHover={() => {}}
            type={PlaceCardType.DEFAULT}
            isAuth={true}
          />
        </BrowserRouter>
    ).toJSON();

    expect(card).toMatchSnapshot();
  });
});
