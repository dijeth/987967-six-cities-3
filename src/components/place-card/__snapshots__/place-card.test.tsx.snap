// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceCard /> should be render correctly with isFavorite === false 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="picture-1"
        width={260}
      />
    </a>
  </div>
  <div
    className=" place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          123
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button  button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/id-1"
        onClick={[Function]}
      >
        title-1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;

exports[`<PlaceCard /> should be render correctly with isPremium === true 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="picture-1"
        width={260}
      />
    </a>
  </div>
  <div
    className=" place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          123
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          From bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/id-1"
        onClick={[Function]}
      >
        title-1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;

exports[`<PlaceCard /> should be render correctly with type === "DEFAULT" 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="picture-1"
        width={260}
      />
    </a>
  </div>
  <div
    className=" place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          123
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          From bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/id-1"
        onClick={[Function]}
      >
        title-1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;

exports[`<PlaceCard /> should be render correctly with type === "FAVORITE" 1`] = `
<article
  className="favorites__card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="favorites__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={110}
        src="picture-1"
        width={150}
      />
    </a>
  </div>
  <div
    className="favorites__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          123
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          From bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/id-1"
        onClick={[Function]}
      >
        title-1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;

exports[`<PlaceCard /> should be render correctly with type === "NEARBY" 1`] = `
<article
  className="near-places__card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="picture-1"
        width={260}
      />
    </a>
  </div>
  <div
    className=" place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          123
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          From bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/id-1"
        onClick={[Function]}
      >
        title-1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
